phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2877,8 +2877,12 @@ public boolean isViewReferenced() {
             Collections.reverse(columnMetadata);
             tableMetaData.addAll(columnMetadata);
             String dataTableName = parent == null || tableType == PTableType.VIEW ? null : parent.getTableName().getString();
-            PIndexState defaultCreateState = PIndexState.valueOf(connection.getQueryServices().getConfiguration().
-                            get(INDEX_CREATE_DEFAULT_STATE, QueryServicesOptions.DEFAULT_CREATE_INDEX_STATE));
+            PIndexState defaultCreateState;
+            String defaultCreateStateString = connection.getClientInfo(INDEX_CREATE_DEFAULT_STATE);
+            if (defaultCreateStateString == null) defaultCreateState = PIndexState.valueOf(connection.getQueryServices().getConfiguration().
```

Comment:
Nit: We put braces for if even if single statement in if-else block and the statements inside are on a separate line